### ISSUE TITLE
Problem identifying source line number when when using project with subfolders

### ISSUE THREAD

**Issue by kudresov** [googleapis/cloud-debug-nodejs#862]
# Overview
I have a monorepo setup where each service leaves under a directory, something like:
/services
  /service-a
    /service-1
      /src
        index.ts
    package.json

When I try to debug the code in Stackdriver breakpoints are not hit (I link the github repo). However if I upload just the subdirectory `service-1` line numbers are getting hit. I tried to reproduce this locally by setting up a repo (see below).

I managed to get the same behaviour locally as when running code in GCP (Cloud Run). After digging through the library code it looks like line number for .js file can't be correctly identified, I get -2 as line number when using the whole project as uploaded source and I get correct line number when uploading just subdirectory.

After debugging it locally it looks like the problem is in: 
`source-map-consumer.js:196`
`needle.source = this._findSourceIndex(needle.source);`
needle.source is `"../service-1/src/index.ts"`
the result is: `-1`

It would be also good to get some sort of error message when we weren't able to set a breakpoint as there is absolutely no feedback in GCP UI, which makes it hard to identify the problem in the first place. As well it looks like nested projects should be supported.

## Config
I have tried providing all combinations of the config I could think with `appPathRelativeToRepository` and  `pathResolver` without success.

#### Environment details

  - OS: Mac OS 10.14.5
  - Node.js version: v10.15.3
  - npm version: 6.13.7
  - `@google-cloud/debug-agent` version: 4.2.2

#### Steps to reproduce
  1. clone repo branch: https://github.com/kudresov/gcp-test/tree/feature/local-setup
  2. cd into `service-1`
  3. run `tsc`
  4. run `node dist/index.js`

**Comment by tcrognon** [comment]
I'm having this same problem. I have a subfolder deployed to Cloud Run and my .ts breakpoints work when only the subfolder is loaded in Cloud Debugger, not when the entire repo is loaded. However, breakpoints work on the compiled .js files regardless, so maybe it's a bug in resolving source maps?

**Comment by tahnik** [comment]
I have a similar problem. I use `tsc` to build my typescript to javascript files with source maps. However, when I get an error like the following in Error Reporting:

```
Error: ENOENT: no such file or directory, open '/usr/src/grpc/src/protos/payment.proto'
at Object.openSync (fs.js:462)
at Object.readFileSync (fs.js:364)
at Object.<anonymous> (/usr/src/app/dist/controllers/payment/methods.js:17)
```
and I click on `/usr/src/app/dist/controllers/payment/methods.js:17`, it does take me to line 17, but in my original source file, the line number where the error happened is line 5. So to me it seems like the cloud debugger is not taking source maps into consideration at all.

**Comment by sarthak3248** [comment]
@tahnik It seems you have a path/directory issue! Change the path to your current working directory or by adding path (control panel->System and security->system->advanced settings->enviornment variable ) and then rerun the code by cd to the current one.
hope it  helped you!

**Comment by BurkovskyNikita** [comment]
I have a similar problem. I have a typescript nodejs app under a `server` subfolder in the repository, so the source files are under `server/src`. I use compute engine to run docker container where the working directory is under `app/` and starting point is `app/build/index.js`

In cloud debugger I manually choose the master branch of my repository in Cloud Source Repositories. Placing a breakpoint under `server/src/index.ts` won't do anything and won't show an error in Debugger UI, but removing the `server/` part will make it work.

I also tried configuring `appPathRelativeToRepository` and/or `pathResolver`, but that didn't help. Setting `appPathRelativeToRepository` to `server` or `server/` or `./server/` made the the UI show an error. Setting `pathResolver` similar to what is done in https://github.com/kudresov/gcp-test/tree/feature/local-setup doesn't help.

Is there something obvious that I'm missing?

**Comment by canove** [comment]
Same here: 

I have a monorepo containig this folder three:

-backend
--backend/source-context.json
--/backend/src (typescript files)
--/backend/dist (transpiled files, with sourcemaps)
--/backenc/src/app.ts
-frontend
--/frotend/...
.gitignore

I'm running backend nodejs project on GKE.
I tried with `appPathRelativeToRepository: "backend"`, but it throw an error when i try to create a snapshot:
`A script matching the source file was not found loaded on the debuggee`

I've added `"sourceMap": true` on my `tsconfig.json`, confirmed that map files are being created on dist folder. But it stills didn't work. 

I'm obliviously missing something :(
